server/internal/openapi/extract_speakeasy.go

```diff
@@ -68,147 +68,330 @@ func (p *ToolExtractor) doSpeakeasy(
 
 	doc, err := parseSpeakeasy(ctx, tracer, bytes.NewReader(data))
 	if err != nil {
-		return nil, oops.E(oops.CodeUnexpected, oops.Permanent(err), "error opening openapi document").Log(ctx, logger)
-	}
+		return nil, oops.E(oops.CodeUnexpected, oops.Permanent(err),
+			"error opening openapi document").Log(ctx, logger)
+	}
+
+	var (
+		upgradeStart    = time.Now()
+		upgradeDuration time.Duration
+		upgradeOutcome  *o11y.Outcome
+		upgrade         = func(
```

Comment:
please no anonymous inline function in Go. Code is sliding to the right more and more and we need to reign it in. Closures don't help with readability.